source/request.c

```diff
@@ -69,7 +72,9 @@ char *requestToString(struct request req, int type){
     for(i=0; i< req.headersCount;i++){
         if(req.headers[i].name!=NULL && strlen(req.headers[i].name)>0){
             strcat(returnString, req.headers[i].name);
-            strcat(returnString, ": ");
+            if(i!=0){
```

Comment:
Jeżeli odczyt będzie poprawny (name==NULL dla pierwszego nagłówka) to ten warunek nie jest potrzebny.